helix-contract/contracts/mapping-token/darwinia/MappingERC20.sol

```diff
@@ -84,8 +84,8 @@ contract MappingERC20 is IERC20, Ownable, Initializable {
         _mint(account, amount);
     }
 
-    // only factory contract can burn with the authority from users
-    function burn(address account, uint256 amount) external onlyOwner {
+    function burn(address account, uint256 amount) external {
+        require(account == msg.sender, "ERC20: burn address not self");
```

Comment:
Maybe we should add approve feature like dstoken.

Reply:
```
function burn(address guy, uint wad) public auth stoppable {
        if (guy != msg.sender && _approvals[guy][msg.sender] != uint(-1)) {
            _approvals[guy][msg.sender] = sub(_approvals[guy][msg.sender], wad);
        }

        _balances[guy] = sub(_balances[guy], wad);
        _supply = sub(_supply, wad);
        emit Burn(guy, wad);
    }
```

Reply:
But we did not support auth here, might not easy to use owner and  support auth here.

It is trade off here.